libs/geo/src/main/java/org/elasticsearch/geometry/simplify/SimplificationErrorCalculator.java

```diff
@@ -0,0 +1,189 @@
+                return Math.sqrt(s * da * db * dc);
+            } else {
+                // rounding errors can cause flat triangles to have negative values, leading to NaN areas
+                return 0.0;
+            }
+        }
+
+        private double distance(PointLike a, PointLike b) {
+            return SloppyMath.haversinMeters(a.y(), a.x(), b.y(), b.x());
+        }
+    }
+
+    /**
+     * Calculate the triangle area using geographic coordinates and Herons formula (side lengths) as described at
+     * <a href="https://en.wikipedia.org/wiki/Area_of_a_triangle">Area of a triangle</a>, but scale the area down
+     * by the inverse of the length of the base (left-right), which estimates the height of the triangle.
+     */
+    class TriangleHeightCalculator implements SimplificationErrorCalculator {
+
+        @Override
+        public double calculateError(PointLike left, PointLike middle, PointLike right) {
+            // Calculate side lengths using approximate haversine
+            double a = distance(left, right);
+            double b = distance(right, middle);
+            double c = distance(middle, left);
+            // semi-perimeter
+            double s = 0.5 * (a + b + c);  // Semi-perimeter
+            double da = s - a;
+            double db = s - b;
+            double dc = s - c;
+            if (da >= 0 && db >= 0 && dc >= 0) {
+                // Herons formula, scaled by 2/a to estimate height
+                return 2.0 * Math.sqrt(s * da * db * dc) / a;
+            } else {
+                // rounding errors can cause flat triangles to have negative values, leading to NaN areas
+                return 0.0;
+            }
+        }
+
+        private double distance(PointLike a, PointLike b) {
+            return SloppyMath.haversinMeters(a.y(), a.x(), b.y(), b.x());
+        }
+    }
+
+    /**
+     * Estimate the error as the height of the point above the base, but including support for back-paths
+     * in the sense that of the point to be removed is father from either end than the height, we take that distance instead.
+     * <p>
+     * Rotate all three points such that left-right are a horizontal line on the x-axis. Then the numbers of interest are:
+     * <ol>
+     *     <li>height: y-value of middle point</li>
+     *     <li>deltaL: -min(0, middleX - leftX)</li>
+     *     <li>deltaR: max(0, middleX - rightX)</li>
+     * </ol>
+     * And the final error is: error = max(height, max(deltaL, deltaR))
+     * <p>
+     * This is not a full Frechet error calculation as it does not maintain state of all removed points,
+     * only calculating the error incurred by removal of the current point, as if the current simplified line is
+     * a good enough approximation of the original line. This restriction is currently true of all the
+     * calculations implemented so far.
+     */
+    class SimpleFrechetErrorCalculator implements SimplificationErrorCalculator {
```

Comment:
While this is based on the FrechetError oracle from the Abam et al. paper, it is not the complete calculation, since it only considers the currently simplified geometry, not the impact on the original geometry. The prefix `Simple` is probably not sufficient to cover this difference, so a better name would be `HeightAndBackpathDistanceCalculator`, which describes literally what it does.